### PR TITLE
[chore] update k8s manifests 1.10.0 release

### PR DESCRIPTION
**Pull request by puckpuck**
# Changes

With the Helm chart for 1.10.0 released, we can update the K8s manifests too.

`make generate-kubernetes-manifests` 

## Merge Requirements

For new features contributions please make sure you have completed the following
essential items:

* ~[ ] `CHANGELOG.md` updated to document new feature additions~
* ~[ ] Appropriate documentation updates in the [docs][]~
* ~[ ] Appropriate Helm chart updates in the [helm-charts][]~

Maintainers will not merge until the above have been completed. If you're unsure
which docs need to be changed ping the
[@open-telemetry/demo-approvers](https://github.com/orgs/open-telemetry/teams/demo-approvers).

[docs]: https://opentelemetry.io/docs/demo/
[helm-charts]: https://github.com/open-telemetry/opentelemetry-helm-charts

### CODE REVIEW

File: kubernetes/opentelemetry-demo.yaml [hunk 1]
```diff
@@ -10114,6 +10179,67 @@ spec:
...
+        app.kubernetes.io/component: imageprovider
+        app.kubernetes.io/name: opentelemetry-demo-imageprovider
+    spec:
+      serviceAccountName: opentelemetry-demo
+      containers:
+        - name: imageprovider
+          image: 'ghcr.io/open-telemetry/demo:1.10.0-imageprovider'
+          imagePullPolicy: IfNotPresent
+          ports:
+
```
Comment (rogercoll) [inline kubernetes/opentelemetry-demo.yaml:10213]: nit: new deployment included, should we add a changelog?

**Review by rogercoll — APPROVED** [review]

File: kubernetes/opentelemetry-demo.yaml (hunk 1 above)
Comment (puckpuck) [reply kubernetes/opentelemetry-demo.yaml:10213]: Updating this file is a lagging change for the 1.10.0 release. I'm not sure we should add a changelog entry for something captured as part of that release itself.

Comment (puckpuck) [reply kubernetes/opentelemetry-demo.yaml:10213]: This does bring up an interesting point about the post-release tasks we need to do. I could certainly see some improvements in tracking and getting those done.

**Review by julianocosta89 — APPROVED** [review]

File: kubernetes/opentelemetry-demo.yaml (hunk 1 above)
Comment (julianocosta89) [reply kubernetes/opentelemetry-demo.yaml:10213]: I wonder if we couldn't have a GH action that builds it and opens a PR whenever there is a new helm release. 🤔

**Pull request merged**